mne/decoding/gsoc.py

```diff
@@ -0,0 +1,195 @@
+# Authors: Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+from .mixin import TransformerMixin
+from ..epochs import _BaseEpochs
+
+
+class EpochsTransformerMixin(TransformerMixin):
+    """Mixin class for reshaping data to Epoch's standard shape
+
+    This class is meant to be inherited by transformers meant to be
+    used in scikit-learn pipeline. Useful utility for making data
+    compatible with mne internal functions.
+
+    Parameters
+    ----------
+    n_chan : int | None
+
+        The number of channels.
+    """
+
+    def __init__(self, n_chan=None):
+        self.n_chan = n_chan
+        self._check_init()
+
+    def fit(self, X, y=None):
+        """No use here. Added for scikit-learn compatibility.
+
+        Parameters
+        ----------
+        X : numpy array of dimensions [2,3,4]
+            The data to be reshaped.
+        y : None
+            Used for scikit-learn compatibilty
+
+        Returns
+        -------
+        self : Instance of EpochsTransformerMixin
+            Return the same object.
+        """
+        return self
+
+    def fit_transform(self, X, y=None):
+        """No use here. Added for scikit-learn compatibility.
+
+        Parameters
+        ----------
+        X : numpy array of dimensions [2,3,4]
+            The data to be reshaped.
+        y : None
+            Used for scikit-learn compatibilty
+
+        Returns
+        -------
+        X : numpy ndarray
+            The same array
+        """
+        self.fit(X, y)
+        return self.transform(X)
+
+    def transform(self, X):
+        """No use here. Added for scikit-learn compatibility.
+
+        Parameters
+        ----------
+        X : numpy array of dimensions [2,3,4]
+            The data to be reshaped.
+
+        Returns
+        -------
+        X : numpy ndarray
+            The same array
+        """
+        return X
+
+    def _reshape(self, X):
+        """Recontruct epochs to get a n_trials * n_chan * n_time
+
+        Parameters
+        ----------
+        X : numpy array of dimensions [2,3,4]
```

Comment:
`X : numpy array, shape(n_epoch, n_chan, n_time) or (n_epoch, n_chan * n_time)`

EpochsTransformerMixin will be specifc to Epochs which can't have frequencies. We'll make a specific class for that one when we'll need it (e.g. PSDTransformerMixin)